runtime/src/sorted_storages.rs

```diff
@@ -45,21 +45,25 @@ impl<'a> SortedStorages<'a> {
                 storage.slot() // this must be unique. Will be enforced in new_with_slots
             })
             .collect::<Vec<_>>();
-        Self::new_with_slots(source, &slots)
+        Self::new_with_slots(source, &slots, None)
     }
 
     // source[i] is in slot slots[i]
     // assumptions:
     // 1. slots vector contains unique slot #s.
     // 2. slots and source are the same len
-    pub fn new_with_slots(source: &'a [SnapshotStorage], slots: &[Slot]) -> Self {
+    pub fn new_with_slots(
```

Comment:
nit: maybe rename to `new_from_storage_entries`?

Reply:
some callers have only &[SnapshotStorage]. We have to figure out the corresponding slots. Restoring from a snapshot is such a code path. Those callers call 'new'.
Other callers can easily generate a corresponding Slot per storage. Those callers call 'new_with_slots'.
We can rename either of them. This is the brief history of the naming.